packages/twenty-front/src/modules/prefetch/components/PrefetchRunQueriesEffect.tsx

```diff
@@ -0,0 +1,98 @@
+import { useEffect } from 'react';
+import { useQuery } from '@apollo/client';
+import { useSetRecoilState } from 'recoil';
+
+import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useUpsertFindManyRecordsQueryInCache } from '@/object-record/cache/hooks/useUpsertFindManyRecordsQueryInCache';
+import { useGenerateFindManyRecordsForMultipleMetadataItemsQuery } from '@/object-record/hooks/useGenerateFindManyRecordsForMultipleMetadataItemsQuery';
+import { useMapConnectionToRecords } from '@/object-record/hooks/useMapConnectionToRecords';
+import { MultiObjectRecordQueryResult } from '@/object-record/relation-picker/hooks/useMultiObjectRecordsQueryResultFormattedAsObjectRecordForSelectArray';
+import { ALL_FAVORITES_QUERY_KEY } from '@/prefetch/query-keys/AllFavoritesQueryKey';
+import { ALL_VIEWS_QUERY_KEY } from '@/prefetch/query-keys/AllViewsQueryKey';
+import { prefetchIsLoadedFamilyState } from '@/prefetch/states/prefetchIsLoadedFamilyState';
+import { PrefetchKey } from '@/prefetch/types/PrefetchKeys';
+import { isDefined } from '~/utils/isDefined';
+
+export const PrefetchRunQueriesEffect = () => {
```

Comment:
Seems like this could be abstracted with those parameters : 

```
{
  objectMetadataForPrefetch: {
    objectNameSingular: string;
    prefetchQueryKey: string; // Or QueryKey ?
  }[]
  depth: number
}
```